### PR TITLE
real match OakSpeechNidoranFSetup

### PR DESCRIPTION
**Pull request by jiangzhengwenjz**

### CODE REVIEW

File: src/pokemon.c [hunk 1]
```diff
@@ -6134,8 +6134,7 @@ struct OakSpeechNidoranFStruct *OakSpeechNidoranFSetup(u8 battlePosition, bool8
             battlePosition = 1;
         if (battlePosition > 8)
             battlePosition = 8;
-        // The following two statements refused to cooperate. 
-        sOakSpeechNidoranResources->spriteCount = battlePosition;
+        sOakSpeechNidoranResources->spriteCount = (battlePosition << 16) >> 16;
```
Comment (PikalaxALT) [inline src/pokemon.c]: this is a fake match

Comment (ProjectRevoTPP) [reply src/pokemon.c]: This is not a fake match. An ameteur programmer can write this.

Comment (huderlem) [reply src/pokemon.c]: This is 100% not what the original programmer wrote.  It's better than no match, though.

Comment (ProjectRevoTPP) [reply src/pokemon.c]: Its not 100%. It's more like 95%. Once again, an ameteur can write this.

**Pull request merged**